libcontainerd/supervisor/remote_daemon.go

```diff
@@ -64,13 +63,10 @@ type Daemon interface {
 type DaemonOpt func(c *remote) error
 
 // Start starts a containerd daemon and monitors it
-func Start(ctx context.Context, rootDir, stateDir string, opts ...DaemonOpt) (Daemon, error) {
+func Start(ctx context.Context, stateDir string, opts ...DaemonOpt) (Daemon, error) {
 	r := &remote{
-		stateDir: stateDir,
 		Config: config.Config{
 			Version: 2,
-			Root:    filepath.Join(rootDir, "daemon"),
-			State:   filepath.Join(stateDir, "daemon"),
```

Comment:
I just noticed that we may have an issue here; it looks like `libcontainerd` _client_ has these paths effectively hard-coded to be inside `/var/run/docker/containerd/`; at least, I see no code that checks what containerd's directory is for those.

https://github.com/moby/moby/blob/6e213899177faac5c4bf9bdc68ac2899fb0916e1/daemon/daemon_unix.go#L1738

https://github.com/moby/moby/blob/49f021ebf00a76d74f5ce158244083e2dfba26fb/daemon/daemon_windows.go#L651



https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/libcontainerd/remote/client_linux.go#L61


https://github.com/moby/moby/blob/5de77049db9d6aed893f086fc672cb8df4532650/libcontainerd/remote/client.go#L904-L906

For plugins (linux) it uses `/run/docker/plugins`; https://github.com/moby/moby/blob/a71463be98b880aec7b6d73690bee994ec7042f0/daemon/daemon_linux.go#L23-L25

For windows, something under the daemon's "root" directory (which is confusing, as "root" (data) is used as "exec" (runtime) path; https://github.com/moby/moby/blob/a71463be98b880aec7b6d73690bee994ec7042f0/daemon/daemon_windows.go#L58-L60

Reply:
Oh, I guess part of this is addressed in https://github.com/rumpl/moby/pull/40, but we probably still need to look for the "not-containerd as snapshotter" situation with a custom configuration? 🤔